blockToAddresses.go

```diff
@@ -23,40 +23,23 @@ type JSONPayload struct {
 
 // Returns all traces for a given block
 func getTracesForBlock(blockNum int) ([]byte, error) {
-    hexBlockNum := fmt.Sprintf("0x%x", blockNum)
-
-    data := JSONPayload {
-        "2.0",
-        "trace_block",
-        Params{hexBlockNum},
-        2,
-    }
-
-    payloadBytes, err := json.Marshal(data)
-    if err != nil {
-        return nil, err
-    }
-
-    body := bytes.NewReader(payloadBytes)
-
-    req, err := http.NewRequest("POST", "http://localhost:8545", body)
-    if err != nil {
-        return nil, err
-    }
-    req.Header.Set("Content-Type", "application/json")
-
-    resp, err := http.DefaultClient.Do(req)
-    if err != nil {
-        return nil, err
-    }
-
-    tracesBody, err := ioutil.ReadAll(resp.Body)
-    if err != nil {
-        return nil, err
+    payloadBytes, err := json.Marshal(JSONPayload{"2.0", "trace_block", Params{fmt.Sprintf("0x%x", blockNum)}, 2})
```

Comment:
I appreciate the lines saved, but Go style is to check and return each error individually. I kinda agree -- I think the code readability suffers with this sort of orginization.